File: Java/Calculator/Calculator.java
// Created by : Samarth Bhandari
// Github : github.com/b-samarth


package Calculator;

import java.lang.*;
import java.util.Scanner;

public class Calculator {

    public static double add(double num1, double num2) {
        return num1 + num2;
    }

    public static double subtract(double num1, double num2) {
        return num1 - num2;
    }

    public static double divide(double num1, double num2) {
        return num1 / num2;
    }

    public static double multiply(double num1, double num2) {
        return num1 * num2;
    }

    public static double modulus(double num1, double num2) {
        return num1 % num2;
    }


    public static int factorial(int num) {
        if (num == 0)
            return 1;
        else 
            return num * factorial(num-1);
    }

    public static double exponent(double num1, double num2) {
        return Math.pow(num1,num2);
    }

    public static double squareroot(double num) {
        return Math.pow(num,0.5);
    }
    
    public static double nthroot(double num1, double num2) {
        return Math.pow(num1, (1 / num2));
    }

    public static String root(double num1, double num2, double num3) {
        double d = Math.pow(num2,2) - (4 * num1 * num3);
        double r1,r2,i;
        String sb = "The roots are ";
        if (d == 0) {
            r1 = num2/(-2 * num1);
            r2 = r1;
            sb = sb + "equal, they are : " + r1 + " and " + r2 + ".";
        }

        else if (d < 0) {
            r1 = num2/(-2 * num1);
            r2 = r1;
            i = Math.pow(-d,0.5) / (2 * num1);
            sb = sb + r1 + i + "i and " + r2 + i + "i.";
        }
        else if (d > 0) {
            r1 = -num2 + Math.pow(d,0.5) / (2 * num1);
            r2 = -num2 - Math.pow(d,0.5) / (2 * num1);
            sb = sb + r1 + " and " + r2 + ".";
        }
        return sb;
    }

    public static double naturalLog(double num) {
        return Math.log(num);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int choice;
        double num1, num2, num3;
        int a;
        System.out.println("=============================================================================");
        System.out.println("1. Addition");
        System.out.println("2. Subtraction");
        System.out.println("3. Multiplication");
        System.out.println("4. Division");
        System.out.println("5. Modulus");
        System.out.println("6. Factorial");
        System.out.println("7. Power");
        System.out.println("8. Square Root");
        System.out.println("9. nth root");
        System.out.println("10. Roots of quadratic equation");
        System.out.println("11. Natural Logarithm");
        System.out.print("Type the number with respect to your desired operation : ");
        choice = sc.nextInt();
        switch (choice) {
            case 1:
                System.out.println("You chose Addition");
                System.out.print("Enter first number : ");
                num1 = sc.nextDouble();
                System.out.print("Enter second number : ");
                num2 = sc.nextDouble();
                System.out.println("The result is : " + add(num1, num2));
                break;
            case 2:
                System.out.println("You chose Subtraction");
                System.out.print("Enter first number : ");
                num1 = sc.nextDouble();
                System.out.print("Enter second number : ");
                num2 = sc.nextDouble();
                System.out.println("The result is : " + subtract(num1, num2));
                break;
            case 3:
                System.out.println("You chose Multiplication");
                System.out.print("Enter first number : ");
                num1 = sc.nextDouble();
                System.out.print("Enter second number : ");
                num2 = sc.nextDouble();
                System.out.println("The result is : " + multiply(num1, num2));
                break;
            case 4:
                System.out.println("You chose Division");
                System.out.print("Enter first number : ");
                num1 = sc.nextDouble();
                System.out.print("Enter second number : ");
                num2 = sc.nextDouble();
                System.out.println("The result is : " + divide(num1, num2));
                break;
            case 5:
                System.out.println("You chose Modulus");
                System.out.print("Enter first number : ");
                num1 = sc.nextDouble();
                System.out.print("Enter second number : ");
                num2 = sc.nextDouble();
                System.out.println("The result is : " + modulus(num1, num2));
                break;
            case 6:
                System.out.println("You chose Factorial");
                System.out.print("Enter the number : ");
                a = sc.nextInt();
                System.out.println("The result is : " + factorial(a));
                break;
            case 7:
                System.out.println("You chose Power");
                System.out.print("Enter first number : ");
                num1 = sc.nextDouble();
                System.out.print("Enter second number : ");
                num2 = sc.nextDouble();
                System.out.println("The result is : " + exponent(num1, num2));
                break;
            case 8:
                System.out.println("You chose Square Root");
                System.out.print("Enter the number : ");
                num1 = sc.nextDouble();
                System.out.println("The result is : " + squareroot(num1));
                break;
            case 9:
                System.out.println("You chose nth root");
                System.out.print("Enter first number : ");
                num1 = sc.nextDouble();
                System.out.print("Enter second number : ");
                num2 = sc.nextDouble();
                System.out.println("The result is : " + nthroot(num1, num2));
                break;
            case 10:
                System.out.println("You chose Roots of Quadratic Equation");
                System.out.print("Enter power of x^2 : ");
                num1 = sc.nextDouble();
                System.out.print("Enter power of x : ");
                num2 = sc.nextDouble();
                System.out.print("Enter constant value : ");
                num3 = sc.nextDouble();
                System.out.println(root(num1, num2, num3));
                break;
            case 11:
                System.out.println("You chose Natural Logarithm");
                System.out.print("Enter the number : ");
                num1 = sc.nextDouble();
                System.out.println("The result is : " + naturalLog(num1));
                break;
            default:
                System.out.println("Invalid Choice, Please Try Again!");
                try {
                    Thread.sleep(2000);
                } catch (Exception e) {
                    e.printStackTrace();
                }

                main(null);
                break;
        }
        System.out.println("=============================================================================");
        
    }
}

File: C/Calculator/Calculator.c
//By Shreya Bera
//https://github.com/shreya-bera
#include<stdio.h>
#include<math.h>
int main()
{
    float num1,num2, result=1,total=0;
    float a, b, c, discriminant, root1, root2, realPart, imagPart,number;
    int num3,op, modulus,x,y, k=0,n;
    printf("\n*******WELCOME TO C CALCULATOR*******\n ");
    printf("\n**************MENU**************\n");
    printf("1. Addition \n");
    printf("2. Subtraction \n");
    printf("3. Multiplication \n");
    printf("4. Division \n");
    printf("5. Modulus\n");
    printf("6. Power \n");
    printf("7. Factorial \n");
    printf("8. Square root \n");
    printf("9. nth root \n");
    printf("10. Natural Logarithm \n");
    printf("11. Roots of quadratic equation \n");
    printf("\nEnter the Number of the operation to be performed:\n ");
    scanf("%d",&op);
    switch(op)
    {
    case 1:
        printf("\nEnter the number of elements you want to add:");
        scanf("%d",&n);
        printf("Please enter %d numbers one by one: \n",n);
        while(k<n)
        {
            scanf("%f",&number);
            total=total+number;
            k=k+1;
        }
        printf("Sum of %d numbers = %.2f \n",n,total);
        break;
    case 2:
        printf("Enter the Numbers:\n");
        scanf("%f%f",&num1,&num2);
        result=num1-num2;
        printf("The result is%.2f",result);
        break;
    case 3:
        printf("Enter the Numbers:\n");
        scanf("%f%f",&num1,&num2);
        result=num1*num2;
        break;
    case 4:
        printf("Enter the Numbers:\n");
        scanf("%f%f",&num1,&num2);
        result=num1/num2;
        printf("The result is%.2f",result);
        break;
    case 5:
        printf("Enter the Numbers:\n");
        scanf("%d%d",&x,&y);
        modulus =x%y;
        printf("The result is %d",modulus);
        break;
    case 6:
        printf("\nEnter two numbers to find the power \n");
        printf("Number: ");
        scanf("%f",&num1);
        printf("Power : ");
        scanf("%f",&num2);
        result=pow(num1,num2);
        printf("The result is%.2f",result);
        break;
    case 7:
        if(num3>0)
        {
            printf("\nEnter the number to find its factorial\n");
            printf("Number:");
            scanf("%d",&num3);
            for(int i=1;i<=num3;i++)
            result=result*i;
            printf("The result is %d",result);
        }
        else
        {
            printf("\nFor finding factorial the number should be positive and integer value.\n ");
        }
        break;
    case 8:
        printf("\nEnter the number to find it's square root :x^2 \n");
        printf("Number: ");
        scanf("%f",&num1);
        result=sqrt(num1);
        printf("The result is%.2f",result);
        break;
    case 9:
        printf("\nEnter a number to find n th root");
        printf("Number:");
        scanf("%f",&num1);
        printf("\nEnter a number to set n : ");
        scanf("%f",&num2);
        result = pow(num1,1/num2);
        printf("The result is%.2f",result);
        break;
    case 10:
        printf("\nEnter a number to find its Logarithm value\n");
        printf("Number:");
        scanf("%f",&num1);
        result=log(num1);
        printf("The result is %.2f",result);
        break;
    case 11:
        printf("\nEnter coefficients a, b and c: \n");
        scanf("%f %f %f", &a, &b, &c);

        discriminant = b * b - 4 * a * c;
        discriminant = b * b - 4 * a * c;
        discriminant = b * b - 4 * a * c;
        if (discriminant > 0)
            {
            root1 = (-b + sqrt(discriminant)) / (2 * a);
            root2 = (-b - sqrt(discriminant)) / (2 * a);
            printf("root1 = %.2lf and root2 = %.2lf\n", root1, root2);
            }
        else if (discriminant == 0)
            {
            root1 = root2 = -b / (2 * a);
            printf("root1 = root2 = %.2lf\n", root1);
            }
        else
            {
            realPart = -b / (2 * a);
            imagPart = sqrt(-discriminant) / (2 * a);
            printf("root1 = %.2f+%.2fi and root2 = %.2f-%.2fi\n", realPart, imagPart, realPart, imagPart);
            }
        break;
      }

}
